clippy_dev/src/stderr_length_check.rs

```diff
@@ -0,0 +1,54 @@
+use std::ffi::OsStr;
+use walkdir::WalkDir;
+
+use std::fs::File;
+use std::io::prelude::*;
+
+// The maximum length allowed for stderr files.
+//
+// We limit this because small files are easier to deal with than bigger files.
+const LIMIT: usize = 320;
+
+pub fn check() {
+    let stderr_files = stderr_files();
+    let exceeding_files = exceeding_stderr_files(stderr_files).collect::<Vec<String>>();
+
+    if !exceeding_files.is_empty() {
```

Comment:
Oh the collection is needed because of this. In this case it doesn't matter if we collect inside the function or here. Sorry for that :D

Reply:
I think it's still better to collect at the call site rather than in the function itself, so it's fine :)